microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/MicroProfileValidator.java

```diff
@@ -167,8 +175,46 @@ private void validatePropertyValue(String propertyName, ItemMetadata metadata, P
 	}
 
 	/**
-	 * Returns an error message only if <code>value</code> is an invalid enum for
-	 * the property defined by <code>metadata</code>
+	 * Validates the property value expressions (${other.property}) of the given property.
+	 *
+	 * Checks if the property expression is closed, and if the referenced property exists.
+	 *
+	 * @param property The property to validate
+	 * @param allProperties A list of all the properties defined in the property file and in the project info
+	 */
+	private void validatePropertyValueExpressions(Property property) {
+		if (property.getValue() == null) {
+			return;
+		}
+		for (Node child : property.getValue().getChildren()) {
+			if (child != null && child.getNodeType() == NodeType.PROPERTY_VALUE_EXPRESSION) {
+				PropertyValueExpression propValExpr = (PropertyValueExpression) child;
+				if (allProperties == null) {
+					// Collect names of all properties defined in the configuration file and the project information
+					allProperties = new HashSet<>();
+					allProperties.addAll(projectInfo.getProperties().stream().map((ItemMetadata info) -> {
+						return info.getName();
+					}).collect(Collectors.toList()));
+					allProperties.addAll(property.getOwnerModel().getChildren().stream().filter(n -> {
+						return n.getNodeType() == NodeType.PROPERTY;
+					}).map(prop -> {
+						return ((Property)prop).getPropertyNameWithProfile();
+					}).collect(Collectors.toList()));
+				}
+				String refdProp = propValExpr.getReferencedPropertyName();
+				if (!propValExpr.isClosed()) {
+					addDiagnostic("Missing '}'", propValExpr, DiagnosticSeverity.Error, ValidationType.missingCloseBrace.name());
+				}
+				if (!allProperties.contains(refdProp)) {

```

Comment:
Severity must be get with :

`DiagnosticSeverity severity = validationSettings.getSyntax().getDiagnosticSeverity(propertyName);`

You should test at first if severity is != none to do the validation (to avoid doing validation for nothing). See validateSyntaxProperty method for instance.